File: lib/screens/trends_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter_screenutil/flutter_screenutil.dart';
import 'package:e_snapp_mobile/utils/constants.dart';

class TrendsScreen extends StatelessWidget {
  const TrendsScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: AppColors.background,
      appBar: AppBar(
        title: Text(
          'Energy Trends',
          style: TextStyle(
            fontSize: 18.sp,
            fontWeight: FontWeight.w600,
            color: AppColors.textPrimary,
          ),
        ),
        backgroundColor: Colors.white,
        elevation: 0,
        iconTheme: IconThemeData(color: AppColors.textPrimary),
      ),
      body: SafeArea(
        child: SingleChildScrollView(
          padding: EdgeInsets.all(16.w),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              // Time Period Selector
              _buildTimePeriodSelector(),
              SizedBox(height: 24.h),
              
              // Energy Consumption Chart
              _buildEnergyChart(),
              SizedBox(height: 24.h),
              
              // Cost Analysis
              _buildCostAnalysis(),
              SizedBox(height: 24.h),
              
              // Usage Patterns
              _buildUsagePatterns(),
            ],
          ),
        ),
      ),
    );
  }

  Widget _buildTimePeriodSelector() {
    return Container(
      padding: EdgeInsets.all(16.w),
      decoration: BoxDecoration(
        color: Colors.white,
        borderRadius: BorderRadius.circular(12),
        border: Border.all(color: AppColors.border),
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(
            'Time Period',
            style: TextStyle(
              fontSize: 16.sp,
              fontWeight: FontWeight.w600,
              color: AppColors.textPrimary,
            ),
          ),
          SizedBox(height: 12.h),
          Row(
            children: [
              Expanded(
                child: _buildPeriodButton('Day', true),
              ),
              SizedBox(width: 8.w),
              Expanded(
                child: _buildPeriodButton('Week', false),
              ),
              SizedBox(width: 8.w),
              Expanded(
                child: _buildPeriodButton('Month', false),
              ),
              SizedBox(width: 8.w),
              Expanded(
                child: _buildPeriodButton('Year', false),
              ),
            ],
          ),
        ],
      ),
    );
  }

  Widget _buildPeriodButton(String period, bool isSelected) {
    return Container(
      padding: EdgeInsets.symmetric(vertical: 8.h, horizontal: 12.w),
      decoration: BoxDecoration(
        color: isSelected ? AppColors.primary : Colors.transparent,
        borderRadius: BorderRadius.circular(8),
        border: Border.all(
          color: isSelected ? AppColors.primary : AppColors.border,
        ),
      ),
      child: Text(
        period,
        style: TextStyle(
          fontSize: 14.sp,
          fontWeight: FontWeight.w500,
          color: isSelected ? Colors.white : AppColors.textSecondary,
        ),
        textAlign: TextAlign.center,
      ),
    );
  }

  Widget _buildEnergyChart() {
    return Container(
      padding: EdgeInsets.all(16.w),
      decoration: BoxDecoration(
        color: Colors.white,
        borderRadius: BorderRadius.circular(12),
        border: Border.all(color: AppColors.border),
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(
            'Energy Consumption',
            style: TextStyle(
              fontSize: 16.sp,
              fontWeight: FontWeight.w600,
              color: AppColors.textPrimary,
            ),
          ),
          SizedBox(height: 16.h),
          Container(
            height: 200.h,
            decoration: BoxDecoration(
              color: AppColors.background,
              borderRadius: BorderRadius.circular(8),
            ),
            child: Center(
              child: Column(
                mainAxisAlignment: MainAxisAlignment.center,
                children: [
                  Icon(
                    Icons.show_chart,
                    size: 48.w,
                    color: AppColors.textSecondary,
                  ),
                  SizedBox(height: 8.h),
                  Text(
                    'Chart will be displayed here',
                    style: TextStyle(
                      fontSize: 14.sp,
                      color: AppColors.textSecondary,
                    ),
                  ),
                ],
              ),
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildCostAnalysis() {
    return Container(
      padding: EdgeInsets.all(16.w),
      decoration: BoxDecoration(
        color: Colors.white,
        borderRadius: BorderRadius.circular(12),
        border: Border.all(color: AppColors.border),
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(
            'Cost Analysis',
            style: TextStyle(
              fontSize: 16.sp,
              fontWeight: FontWeight.w600,
              color: AppColors.textPrimary,
            ),
          ),
          SizedBox(height: 16.h),
          Row(
            children: [
              Expanded(
                child: _buildCostCard(
                  'Total Cost',
                  '\$245.67',
                  AppColors.primary,
                  Icons.attach_money,
                ),
              ),
              SizedBox(width: 12.w),
              Expanded(
                child: _buildCostCard(
                  'Average/Day',
                  '\$8.19',
                  AppColors.success,
                  Icons.trending_up,
                ),
              ),
            ],
          ),
          SizedBox(height: 12.h),
          Row(
            children: [
              Expanded(
                child: _buildCostCard(
                  'Peak Usage',
                  '\$12.45',
                  AppColors.warning,
                  Icons.warning,
                ),
              ),
              SizedBox(width: 12.w),
              Expanded(
                child: _buildCostCard(
                  'Lowest Usage',
                  '\$3.21',
                  AppColors.accent,
                  Icons.trending_down,
                ),
              ),
            ],
          ),
        ],
      ),
    );
  }

  Widget _buildCostCard(String title, String value, Color color, IconData icon) {
    return Container(
      padding: EdgeInsets.all(12.w),
      decoration: BoxDecoration(
        color: color.withOpacity(0.1),
        borderRadius: BorderRadius.circular(8),
        border: Border.all(color: color.withOpacity(0.3)),
      ),
      child: Column(
        children: [
          Icon(
            icon,
            size: 24.w,
            color: color,
          ),
          SizedBox(height: 8.h),
          Text(
            title,
            style: TextStyle(
              fontSize: 12.sp,
              color: AppColors.textSecondary,
            ),
            textAlign: TextAlign.center,
          ),
          SizedBox(height: 4.h),
          Text(
            value,
            style: TextStyle(
              fontSize: 16.sp,
              fontWeight: FontWeight.bold,
              color: color,
            ),
            textAlign: TextAlign.center,
          ),
        ],
      ),
    );
  }

  Widget _buildUsagePatterns() {
    return Container(
      padding: EdgeInsets.all(16.w),
      decoration: BoxDecoration(
        color: Colors.white,
        borderRadius: BorderRadius.circular(12),
        border: Border.all(color: AppColors.border),
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(
            'Usage Patterns',
            style: TextStyle(
              fontSize: 16.sp,
              fontWeight: FontWeight.w600,
              color: AppColors.textPrimary,
            ),
          ),
          SizedBox(height: 16.h),
          _buildPatternItem(
            'Morning (6AM-12PM)',
            '35%',
            AppColors.primary,
          ),
          SizedBox(height: 8.h),
          _buildPatternItem(
            'Afternoon (12PM-6PM)',
            '25%',
            AppColors.accent,
          ),
          SizedBox(height: 8.h),
          _buildPatternItem(
            'Evening (6PM-12AM)',
            '30%',
            AppColors.warning,
          ),
          SizedBox(height: 8.h),
          _buildPatternItem(
            'Night (12AM-6AM)',
            '10%',
            AppColors.success,
          ),
        ],
      ),
    );
  }

  Widget _buildPatternItem(String time, String percentage, Color color) {
    return Row(
      children: [
        Container(
          width: 12.w,
          height: 12.w,
          decoration: BoxDecoration(
            color: color,
            shape: BoxShape.circle,
          ),
        ),
        SizedBox(width: 12.w),
        Expanded(
          child: Text(
            time,
            style: TextStyle(
              fontSize: 14.sp,
              color: AppColors.textPrimary,
            ),
          ),
        ),
        Text(
          percentage,
          style: TextStyle(
            fontSize: 14.sp,
            fontWeight: FontWeight.w600,
            color: color,
          ),
        ),
      ],
    );
  }
} 

File: src/screens/TrendsScreen.js
import React, {useState, useEffect} from 'react';
import {
  View,
  Text,
  StyleSheet,
  ScrollView,
  TouchableOpacity,
  StatusBar,
} from 'react-native';
import {COLORS, FONTS, SIZES, SPACING, TIME_PERIODS} from '../constants';
import {useEnergy} from '../context/EnergyContext';

const TrendsScreen = () => {
  const {energyData, isLoading} = useEnergy();
  const [selectedPeriod, setSelectedPeriod] = useState(TIME_PERIODS.week);

  const periods = [
    {key: TIME_PERIODS.day, label: 'Day'},
    {key: TIME_PERIODS.week, label: 'Week'},
    {key: TIME_PERIODS.month, label: 'Month'},
    {key: TIME_PERIODS.year, label: 'Year'},
  ];

  const renderHeader = () => (
    <View style={styles.header}>
      <Text style={styles.title}>Energy Trends</Text>
      <Text style={styles.subtitle}>Monitor your energy consumption patterns</Text>
    </View>
  );

  const renderPeriodSelector = () => (
    <View style={styles.periodSelector}>
      {periods.map(period => (
        <TouchableOpacity
          key={period.key}
          style={[
            styles.periodButton,
            selectedPeriod === period.key && styles.periodButtonActive,
          ]}
          onPress={() => setSelectedPeriod(period.key)}>
          <Text
            style={[
              styles.periodButtonText,
              selectedPeriod === period.key && styles.periodButtonTextActive,
            ]}>
            {period.label}
          </Text>
        </TouchableOpacity>
      ))}
    </View>
  );

  const renderChartPlaceholder = () => (
    <View style={styles.chartContainer}>
      <View style={styles.chartPlaceholder}>
        <Text style={styles.chartIcon}>📊</Text>
        <Text style={styles.chartTitle}>Energy Consumption Chart</Text>
        <Text style={styles.chartSubtitle}>
          Chart will be displayed here for {selectedPeriod} period
        </Text>
      </View>
    </View>
  );

  const renderStatsCards = () => (
    <View style={styles.statsContainer}>
      <View style={styles.statsRow}>
        <View style={styles.statCard}>
          <Text style={styles.statIcon}>⚡</Text>
          <Text style={styles.statValue}>12.5 kWh</Text>
          <Text style={styles.statLabel}>Total Usage</Text>
        </View>
        <View style={styles.statCard}>
          <Text style={styles.statIcon}>💰</Text>
          <Text style={styles.statValue}>$45.20</Text>
          <Text style={styles.statLabel}>Total Cost</Text>
        </View>
      </View>
      <View style={styles.statsRow}>
        <View style={styles.statCard}>
          <Text style={styles.statIcon}>📈</Text>
          <Text style={styles.statValue}>+8.5%</Text>
          <Text style={styles.statLabel}>vs Last Period</Text>
        </View>
        <View style={styles.statCard}>
          <Text style={styles.statIcon}>🌱</Text>
          <Text style={styles.statValue}>2.3 kg</Text>
          <Text style={styles.statLabel}>CO2 Saved</Text>
        </View>
      </View>
    </View>
  );

  const renderUsageBreakdown = () => (
    <View style={styles.breakdownContainer}>
      <Text style={styles.sectionTitle}>Usage Breakdown</Text>
      <View style={styles.breakdownCard}>
        <View style={styles.breakdownItem}>
          <View style={styles.breakdownIconContainer}>
            <Text style={styles.breakdownIcon}>🏠</Text>
          </View>
          <View style={styles.breakdownContent}>
            <Text style={styles.breakdownTitle}>Living Room</Text>
            <Text style={styles.breakdownSubtitle}>4.2 kWh (33.6%)</Text>
          </View>
          <View style={styles.breakdownBar}>
            <View style={[styles.breakdownBarFill, {width: '33.6%'}]} />
          </View>
        </View>
        <View style={styles.breakdownItem}>
          <View style={styles.breakdownIconContainer}>
            <Text style={styles.breakdownIcon}>🍳</Text>
          </View>
          <View style={styles.breakdownContent}>
            <Text style={styles.breakdownTitle}>Kitchen</Text>
            <Text style={styles.breakdownSubtitle}>3.8 kWh (30.4%)</Text>
          </View>
          <View style={styles.breakdownBar}>
            <View style={[styles.breakdownBarFill, {width: '30.4%'}]} />
          </View>
        </View>
        <View style={styles.breakdownItem}>
          <View style={styles.breakdownIconContainer}>
            <Text style={styles.breakdownIcon}>🛏️</Text>
          </View>
          <View style={styles.breakdownContent}>
            <Text style={styles.breakdownTitle}>Bedroom</Text>
            <Text style={styles.breakdownSubtitle}>2.1 kWh (16.8%)</Text>
          </View>
          <View style={styles.breakdownBar}>
            <View style={[styles.breakdownBarFill, {width: '16.8%'}]} />
          </View>
        </View>
        <View style={styles.breakdownItem}>
          <View style={styles.breakdownIconContainer}>
            <Text style={styles.breakdownIcon}>🚿</Text>
          </View>
          <View style={styles.breakdownContent}>
            <Text style={styles.breakdownTitle}>Bathroom</Text>
            <Text style={styles.breakdownSubtitle}>1.4 kWh (11.2%)</Text>
          </View>
          <View style={styles.breakdownBar}>
            <View style={[styles.breakdownBarFill, {width: '11.2%'}]} />
          </View>
        </View>
        <View style={styles.breakdownItem}>
          <View style={styles.breakdownIconContainer}>
            <Text style={styles.breakdownIcon}>🔧</Text>
          </View>
          <View style={styles.breakdownContent}>
            <Text style={styles.breakdownTitle}>Other</Text>
            <Text style={styles.breakdownSubtitle}>1.0 kWh (8%)</Text>
          </View>
          <View style={styles.breakdownBar}>
            <View style={[styles.breakdownBarFill, {width: '8%'}]} />
          </View>
        </View>
      </View>
    </View>
  );

  return (
    <View style={styles.container}>
      <StatusBar
        barStyle="dark-content"
        backgroundColor={COLORS.white}
        translucent={false}
      />
      
      <ScrollView
        style={styles.scrollView}
        contentContainerStyle={styles.scrollContent}
        showsVerticalScrollIndicator={false}>
        {renderHeader()}
        {renderPeriodSelector()}
        {renderChartPlaceholder()}
        {renderStatsCards()}
        {renderUsageBreakdown()}
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: COLORS.background,
  },
  scrollView: {
    flex: 1,
  },
  scrollContent: {
    paddingBottom: SPACING.xl,
  },
  header: {
    paddingHorizontal: SPACING.xl,
    paddingVertical: SPACING.lg,
    backgroundColor: COLORS.white,
    borderBottomWidth: 1,
    borderBottomColor: COLORS.border,
  },
  title: {
    fontSize: 24,
    fontFamily: FONTS.bold,
    color: COLORS.textPrimary,
    marginBottom: SPACING.sm,
  },
  subtitle: {
    fontSize: 16,
    fontFamily: FONTS.regular,
    color: COLORS.textSecondary,
  },
  periodSelector: {
    flexDirection: 'row',
    paddingHorizontal: SPACING.xl,
    paddingVertical: SPACING.lg,
    backgroundColor: COLORS.white,
    borderBottomWidth: 1,
    borderBottomColor: COLORS.border,
  },
  periodButton: {
    flex: 1,
    paddingVertical: SPACING.md,
    paddingHorizontal: SPACING.lg,
    marginHorizontal: SPACING.xs,
    borderRadius: 8,
    alignItems: 'center',
  },
  periodButtonActive: {
    backgroundColor: COLORS.primary,
  },
  periodButtonText: {
    fontSize: 14,
    fontFamily: FONTS.medium,
    color: COLORS.textSecondary,
  },
  periodButtonTextActive: {
    color: COLORS.white,
  },
  chartContainer: {
    paddingHorizontal: SPACING.xl,
    paddingVertical: SPACING.lg,
  },
  chartPlaceholder: {
    backgroundColor: COLORS.white,
    borderRadius: 12,
    padding: SPACING.xl,
    alignItems: 'center',
    borderWidth: 1,
    borderColor: COLORS.border,
  },
  chartIcon: {
    fontSize: 48,
    marginBottom: SPACING.md,
  },
  chartTitle: {
    fontSize: 18,
    fontFamily: FONTS.semiBold,
    color: COLORS.textPrimary,
    marginBottom: SPACING.sm,
  },
  chartSubtitle: {
    fontSize: 14,
    fontFamily: FONTS.regular,
    color: COLORS.textSecondary,
    textAlign: 'center',
  },
  statsContainer: {
    paddingHorizontal: SPACING.xl,
    marginBottom: SPACING.xl,
  },
  statsRow: {
    flexDirection: 'row',
    marginBottom: SPACING.md,
  },
  statCard: {
    flex: 1,
    backgroundColor: COLORS.white,
    borderRadius: 12,
    padding: SPACING.lg,
    marginHorizontal: SPACING.xs,
    alignItems: 'center',
    shadowColor: COLORS.textPrimary,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 3,
  },
  statIcon: {
    fontSize: 24,
    marginBottom: SPACING.sm,
  },
  statValue: {
    fontSize: 18,
    fontFamily: FONTS.bold,
    color: COLORS.textPrimary,
    marginBottom: SPACING.xs,
  },
  statLabel: {
    fontSize: 12,
    fontFamily: FONTS.regular,
    color: COLORS.textSecondary,
  },
  breakdownContainer: {
    paddingHorizontal: SPACING.xl,
  },
  sectionTitle: {
    fontSize: 18,
    fontFamily: FONTS.bold,
    color: COLORS.textPrimary,
    marginBottom: SPACING.lg,
  },
  breakdownCard: {
    backgroundColor: COLORS.white,
    borderRadius: 12,
    padding: SPACING.lg,
    shadowColor: COLORS.textPrimary,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 3,
  },
  breakdownItem: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: SPACING.lg,
  },
  breakdownIconContainer: {
    width: 40,
    height: 40,
    borderRadius: 8,
    backgroundColor: COLORS.primary + '20',
    justifyContent: 'center',
    alignItems: 'center',
    marginRight: SPACING.md,
  },
  breakdownIcon: {
    fontSize: 20,
  },
  breakdownContent: {
    flex: 1,
  },
  breakdownTitle: {
    fontSize: 14,
    fontFamily: FONTS.semiBold,
    color: COLORS.textPrimary,
    marginBottom: SPACING.xs,
  },
  breakdownSubtitle: {
    fontSize: 12,
    fontFamily: FONTS.regular,
    color: COLORS.textSecondary,
  },
  breakdownBar: {
    width: 60,
    height: 4,
    backgroundColor: COLORS.border,
    borderRadius: 2,
  },
  breakdownBarFill: {
    height: '100%',
    backgroundColor: COLORS.primary,
    borderRadius: 2,
  },
});

export default TrendsScreen; 